Console improvements

permcody opened the pull request:
This supplants #5958. 
- Improves the Console indentation regex so that we don't need all the hackery of stripping and re-adding the newline
- Cleans up unnecessary buffer flushes in the EigenExecutioners
- Removes the last piece of duplicated logic in Multiapp.C from a8a56e49 (Why are you always fillingPositions @friedmud :wink:)

YaqiWang committed:
various improvements on the console outputs for multiapps #5956

permcody committed:
Cleaning up the buffer indent regex and undoing the hackery

refs #5958

permcody committed:
Cleanup execess std::endl calls

permcody commented:
I also removed the Magenta. I'm not sure I want that in every Multiapp simulation. I'm not sure I like the Cyan though either. Perhaps we could make it an option if @YaqiWang really wants the color to help him see the transfers better.

YaqiWang commented:
@permcody I saw one empty line with yak

```
part_solve10: ************************************************************************************
part_solve10: 
part_solve10:                                      YakTest       
part_solve10: 
part_solve10: ************************************************************************************
part_solve10: 
part_solve10: 
part_solve10: 
part_solve10: YAK version:             git commit 5d479d4 on 2015-12-04
part_solve10: 

```

friedmud commented:
@permcody those damn positions MUST be filled ;-)
On Fri, Dec 4, 2015 at 5:27 PM Yaqi <notifications@github.com> wrote:

> @permcody <https://github.com/permcody> I saw one empty line with yak
>
> part_solve10: ************************************************************************************
> part_solve10:
> part_solve10:                                      YakTest
> part_solve10:
> part_solve10: ************************************************************************************
> part_solve10:
> part_solve10:
> part_solve10:
> part_solve10: YAK version:             git commit 5d479d4 on 2015-12-04
> part_solve10:
>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/6021#issuecomment-162099507>.
>

YaqiWang commented:
![ic24340](https://cloud.githubusercontent.com/assets/6607496/11603084/6ff03646-9a9c-11e5-8bfa-907f1e09cb48.jpeg)
how about maroon, color of TAMU, hehe?

jwpeterson commented:
> how about maroon, color of TAMU, hehe?

:astonished:

jwpeterson commented:
@permcody this does not seem to be "under development" any more?

permcody commented:
Yes, I haven't gotten back to it. While it works for all the MOOSE tests, there's still one extra line when Yaqi prints his headers that I need to look at.

permcody committed:
Properly indent headers

This commit adds a newline after the header to handle both blank headers
and headers that end in newlines.

refs #5956

permcody commented:
This PR should be ready for review once more. I added one additional commit to demonstrate the fix for the header problem @YaqiWang illustrated.

YaqiWang commented:
let me take a look really quick.

YaqiWang commented:
Looks good now! We can decide the coloring scheme later.

jwpeterson commented on framework/include/outputs/OutputWarehouse.h:
```diff
@@ -33,7 +33,7 @@ class OutputWarehouse : public Warehouse<Output>
    * Class constructor
    */
-  OutputWarehouse();
+  OutputWarehouse(MooseApp & app);
```
There's a MOOSE in the outhouse.
